sound/soc/sof/pcm.c

```diff
@@ -325,14 +325,13 @@ static int sof_pcm_trigger(struct snd_soc_component *component,
 			ipc_first = true;
 		break;
 	case SNDRV_PCM_TRIGGER_SUSPEND:
-		if (sdev->system_suspend_target == SOF_SUSPEND_S0IX &&
+		/*
+		 * set the suspend_ignored flag for D0I3-compatible capture streams to keep the
+		 * firmware pipeline running when the system is suspended to S0IX.
+		 */
+		if (substream->stream == SNDRV_PCM_STREAM_CAPTURE &&
+		    sdev->system_suspend_target == SOF_SUSPEND_S0IX &&
```

Comment:
If playback streams are suspended by the ALSA framework, aren't capture streams also suspended by the very same framework.

It's as if the 'capture' definitions covered two different cases:
a) Hotwording where there are no data transferred
b) regular capture with periodic DMA transfers.

I just don't see how b) would be handled.

Reply:
@plbossart we dont have a way to distinguish between a D0i3-compatible capture stream and a wov stream in the case of capture today. But we also don't have a regular D0i3-compatible capture stream in our typologies today. I'm not sure how we would handle that if we were to add it.

Reply:
@plbossart updated the patch now to remove the D0i3 during S0ix feature for IPC4 altogether